### PR TITLE
fix: improve cpp I/O library

### PR DESCRIPTION
**Pull request by w43322**
bugfix - [#129](https://github.com/j178/leetgo/issues/129)

moved I/O functions to a separate namespace, removing boilerplate code in lang/cpp.go

add CI for cpp I/O library

### CODE REVIEW

File: testutils/cpp/LC_IO.h [hunk 1]
```diff
@@ -65,122 +26,193 @@ struct TreeNode {
...
-    os << '[';
-    for (auto &&x : v) {
+    /**
+     * Function for printing a variable.
+     */
+    template<typename T>
+    void print(std::ostream &os, const T& x) {
         if constexpr (std::is_same_v<T, std::string>) {
-            os << quoted(x) << ',';
+            os.put('"'); os << x; os.put('"');
```
[the remark this answers is not in the document]
Comment (w43322) [reply testutils/cpp/LC_IO.h:186]: Interesting, I haven't thought of this. I'll change it to the quoted version.
However, does leetcode actually have this kind of input? Also, what about a char `'"'`, do we need to print it as `"\""`?

File: testutils/cpp/ci/LC_IO_CI.cpp
[the remark this answers is not in the document]
Comment (w43322) [reply testutils/cpp/ci/LC_IO_CI.cpp:1]: I renamed the folder. However I'm not familiar with ci setup on GitHub Actions, I might need to learn about it first.

[the remark this answers is not in the document]
Comment (j178) [reply testutils/cpp/ci/LC_IO_CI.cpp:1]: Don’t worry, I’ll do that some time later.

**Review by j178 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**